solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java

```diff
@@ -841,83 +851,139 @@ private CommitVersionInfo getIndexVersion() {
+            "solr_replication_time_elapsed",
+            "Time elapsed during replication in seconds",
+            OtelUnit.SECONDS);
+
+    ObservableLongMeasurement bytesDownloaded =
+        solrMetricsContext.longMeasurement(
+            "solr_replication_bytes_downloaded",
+            "Total bytes downloaded during replication",
+            OtelUnit.BYTES);
+
+    ObservableLongMeasurement downloadSpeed =
+        solrMetricsContext.longMeasurement(
+            "solr_replication_download_speed", "Download speed in bytes per second");
+
+    // Use batch callback to ensure consistent fetcher reference
+    fetcherMetricsBatch =
+        solrMetricsContext.batchCallback(
+            () -> {
               IndexFetcher fetcher = currentIndexFetcher;
               if (fetcher != null) {
-                map.put(LEADER_URL, fetcher.getLeaderCoreUrl());
-                if (getPollInterval() != null) {
-                  map.put(ReplicationAPIBase.POLL_INTERVAL, getPollInterval());
-                }
-                map.put("isPollingDisabled", isPollingDisabled());
-                map.put("isReplicating", isReplicating());
+                isPollingDisabled.record(isPollingDisabled() ? 1 : 0, replicationAttributes);
+                isReplicating.record(isReplicating() ? 1 : 0, replicationAttributes);
+
                 long elapsed = fetcher.getReplicationTimeElapsed();
                 long val = fetcher.getTotalBytesDownloaded();
                 if (elapsed > 0) {
-                  map.put("timeElapsed", elapsed);
-                  map.put("bytesDownloaded", val);
-                  map.put("downloadSpeed", val / elapsed);
+                  timeElapsed.record(elapsed, replicationAttributes);
+                  bytesDownloaded.record(val, replicationAttributes);
+                  downloadSpeed.record(val / elapsed, replicationAttributes);
                 }
-                Properties props = loadReplicationProperties();
-                addReplicationProperties(map::putNoEx, props);
               }
-            });
-    solrMetricsContext.gauge(fetcherMap, true, "fetcher", getCategory().toString(), scope);
-    solrMetricsContext.gauge(
-        () -> isLeader && includeConfFiles != null ? includeConfFiles : "",
-        true,
-        "confFilesToReplicate",
-        getCategory().toString(),
-        scope);
-    solrMetricsContext.gauge(
-        () -> isLeader ? getReplicateAfterStrings() : Collections.<String>emptyList(),
-        true,
-        REPLICATE_AFTER,
-        getCategory().toString(),
-        scope);
-    solrMetricsContext.gauge(
-        () -> isLeader && replicationEnabled.get(),
-        true,
-        "replicationEnabled",
-        getCategory().toString(),
-        scope);
+            },
+            isPollingDisabled,
+            isReplicating,
+            timeElapsed,
+            bytesDownloaded,
+            downloadSpeed);
+  }
+
+  @Override
+  public void close() throws IOException {
```

Comment:
Interesting OTEL difference... gauges need to be closed.  Good to know, and makes sense to me.